popup.js

```diff
@@ -67,18 +69,94 @@ function outputHTMLString(messages) {
  *     scripts.
  * @param {number} i The index of messages to use.
  */
-function fillRemainingHoles(messages, i) {
+function fillRemainingHolesAndMinimizeStyles(messages, i) {
   var html = messages[i].html;
   var frameHoles = messages[i].frameHoles;
   for (var index in frameHoles) {
     if (frameHoles.hasOwnProperty(index)) {
       var frameIndex = frameHoles[index];
       for (var j = 0; j < messages.length; j++) {
         if (messages[j].frameIndex == frameIndex) {
-          fillRemainingHoles(messages, j);
+          fillRemainingHolesAndMinimizeStyles(messages, j);
           html[index] = messages[j].html.join('');
         }
       }
     }
   }
+  minimizeStyles(messages[i]);
+}
+
+/**
+ * Removes all style attribute properties that are unneeded.
+ *
+ * @param{Object} message The message Object whose style attributes should be
+ *     minimized.
+ */
+function minimizeStyles(message) {
+  var nestingDepth = message.frameIndex.split('.').length - 1;
+  var iframe = document.createElement('iframe');
+  document.body.appendChild(iframe);
+  iframe.setAttribute(
+    'style',
+    `height: ${message.windowHeight}px;` + 
+    `width: ${message.windowWidth}px;`
+  );
+  var html = message.html.join('');
+  html = unescapeHTML(html, nestingDepth);
+  iframe.contentDocument.documentElement.innerHTML = html;
+  var doc = iframe.contentDocument;
+
+  for (var id in message.idToStyleIndex) {
+    var index = message.idToStyleIndex[id];
+    var fullStyleDeclaration = message.html[index];
+    var element = doc.getElementById(id);
+    if (element) {
+      element.removeAttribute('style');
+      var unstyledStyle = doc.defaultView.getComputedStyle(element, null);
+      var stylePrefix;
+      var styleSuffix;
+      // Remove style declaration from style attribute.
+      var style = fullStyleDeclaration.replace(
+        /^style=("|&(amp;)*?quot;)(.*)("|&(amp;)*?quot;) ?$/,
```

Comment:
This regex scares me, and I worry that re-parsing the style (here and below) could lead to bugs if the regex isn't perfect. WDYT of using maps instead so we don't need to do this parsing?

Something like this:
```
    if (element) {
      var originalStyle = element.getAttribute('style');
      var styledStyle = doc.defaultView.getComputedStyle(element, null);

      // We need to save off the CSSComputedStyle values because this object is
      // 'live' and removing the style attribute will affect it.
      var styledStyleMap = {};
      for (var propertyIndex = 0; propertyIndex < styledStyle.length; propertyIndex++) {
        var propertyName = styledStyle.item(propertyIndex);
        styledStyleMap[propertyName] = styledStyle.getPropertyValue(propertyName);
      }
      
      element.removeAttribute('style');
      var unstyledStyle = doc.defaultView.getComputedStyle(element, null);
      var neededStyles = '';
      for (var propertyName in styledStyleMap) {
        if (styledStyleMap[propertyName] !== unstyledStyle.getPropertyValue(propertyName))
          neededStyles += `${propertyName}: ${styledStyleMap[propertyName]}; `;
      }

      // FIXME: maybe expose escapedCharacter as a static function?
      // FIXME: this doesn't work with the extension because html serializer is not defined. Maybe import it?
      var escapedStyleQuotes = (new HTMLSerializer()).escapedCharacter('"', nestingDepth+1);
      neededStyles = neededStyles.replace(/"/g, escapedStyleQuotes);
      var quote = (new HTMLSerializer()).escapedCharacter('"', nestingDepth);
      message.html[index] = `style=${quote}${neededStyles}${quote} `;
      element.setAttribute('style', originalStyle);
    }
```